lib/test.js

```diff
@@ -417,6 +418,11 @@ Test.prototype['throws'] = function (fn, expected, msg, extra) {
         expected = String(expected);
     }
 
+    if (expected instanceof Function) {
+        passed = caught.error.constructor === expected;
```

Comment:
@wilmoore that's the semantics of `require('assert')` in node core on purpose.

Having different semantics from `require('assert')` is confusing and not worth it, even if they are "technically" better.